app/screens/home/maximize_screen.tsx

```diff
@@ -124,31 +125,43 @@ export default function MaximizeScreen({
           </ThemedView>
 
           {/* Location button */}
-          <ThemedIconButton
-            name="location-outline"
-            onPress={() => {
-              /* location button */
-            }}
-            size={25}
-            colorType="white"
-          />
+          {challenge.location && (
+            <ThemedIconButton
+              name="location-outline"
+              onPress={() => {
+                navigation.navigate("MapScreen", {
+                  navigation: navigation,
+                  user: user,
+                  firestoreCtrl: firestoreCtrl,
+                  location: {
+                    coords: {
+                      latitude: challenge.location?.latitude,
+                      longitude: challenge.location?.longitude,
+                    },
```

Comment:
Why do you add all the props to the navigate method and not only the required value location ?

Reply:
I was not aware that we used `GeoPoint`s now and continued using `LocationObject`s. This is indeed way clearer and was modified accordingly!